spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/mapping/MappingRuleInterpreterGroupExplode.scala

```diff
@@ -110,6 +144,62 @@ case class MappingRuleInterpreterGroupExplode(rule: MappingConformanceRule,
     errorsDf
   }
 
+  private def flattenOutputsAndAddErrorsAndDefaultsInFlat(outputsStructColumnName: String, frame: DataFrame,
+                                               defaultMappingValues: Map[String, String],
+                                               mappingErrUdfCall: Column,
+                                               errorConditions: Column) = {
+    val errorsApplied = NestedArrayTransformations.nestedWithColumnAndErrorMap(frame, outputsStructColumnName, outputsStructColumnName,
+      ErrorMessage.errorColumnName, c => c, _ => {
+        when(errorConditions, mappingErrUdfCall).otherwise(null) // scalastyle:ignore null
+      }
+    )
+
+    val flattenedColumns = rule.allOutputColumns().map{
+      case (outputColumn, targetAttribute) => {
+        val fieldInOutputs = col(outputsStructColumnName + "." + outputColumn)
+        defaultMappingValues.get(targetAttribute) match {
+          case Some(defValue) => when(fieldInOutputs.isNotNull, fieldInOutputs as outputColumn)
+            .otherwise(expr(defValue)) as outputColumn
+          case None => fieldInOutputs as outputColumn
+        }
```

Comment:
I think this part be extracted to a local def to make the rest of the code more readable.